cmd/minishift/cmd/start_preflight.go

```diff
@@ -414,3 +423,29 @@ func isMounted(driver drivers.Driver, mountpoint string) (bool, error) {
 
 	return true, nil
 }
+
+// checkIsoUrl checks the Iso url and returns true if the iso file exists
+func checkIsoUrl() bool {
+	isoUrl := viper.GetString(configCmd.ISOUrl.Name)
+
+	if isoUrl == constants.B2dIsoAlias || isoUrl == constants.CentOsIsoAlias || isoUrl == constants.MinikubeIsoAlias {
+		return true
+	}
+
+	if runtime.GOOS == "windows" {
```

Comment:
@anjannath , I believe, we can have one quick check of extension before doing OS based file path check. If iso path doesn't contain `.iso` extention, we even don't need to check OS based path.